docker_multiarch/arm.crosscompile.mk

```diff
@@ -115,13 +100,10 @@ USE_LAPACK_PATH =
 USE_INTEL_PATH = NONE
 
 # If use MKL only for BLAS, choose static link automatically to allow python wrapper
-ifeq ($(USE_MKL2017), 0)
+USE_STATIC_MKL = NONE
 ifeq ($(USE_BLAS), mkl)
 USE_STATIC_MKL = 1
```

Comment:
Full MKL supports static lib option, the above flag enables that option. mklml only support dynamic; i believe both support all blas API.

Reply:
Thanks for the clarification, so do you mean USE_BLAS=mkl is used for using full MKL as blas library (and using static link)? If we want to use mklml as blas library (it is more light weighted if it support all blas API), how should we set the compile option? 

I also notice there is some potential conflict here, if set USE_MKLDNN=1 and USE_BLAS=mkl, looks the generated libmxnet.so will both static link to mkl library and dynamic link to mklml library:
Static link to mkl library:
mshadow.mk line 63 
MSHADOW_LDFLAGS += -L${MKLROOT}/../compiler/lib/intel64 -Wl,--start-group ${MKLROOT}/lib/intel64/libmkl_intel_lp64.a ${MKLROOT}/lib/intel64/libmkl_core.a ${MKLROOT}/lib/intel64/libmkl_intel_thread.a -Wl,--end-group -liomp5 -ldl -lpthread -lm

Dynamic link to mklml library:
mshadow.mk line 78: 
MSHADOW_LDFLAGS += -Wl,--as-needed -lmklml_intel -lmklml_gnu 

Is there any potential conflict here? If both MKL and MKLML library could provide the blas API, which library it will use?

Reply:
Please respond to this que

Reply:
Question*

Reply:
@jinhuang415 @marcoabreu 
MSHADOW can use MKLML (which is automatically selected when users build with MKLDNN);  USE_BLAS=mkl is specifically for full mkl (not related to MKLML or MKLDNN). So there is no conflict here; Either of those options support full BLAS compatible API; 

mshadow handles both of these cases already; This PR or MKL-DNN integration PR , do not change this behavior. It is the same as before.

As @zheng-da mentioned, we can modify mshadow to directly use MKL-DNN build flag in another PR. It wasnt done before, as we did not have MKL-DNN integration (we do now). 

This PR scope is limited to removing MKL_EXPERIMENTAL and cleaning up minor make file left overs.